main.cc

```diff
@@ -545,6 +545,94 @@ static locator::host_id initialize_local_info_thread(sharded<db::system_keyspace
     return linfo.host_id;
 }
 
+static bool raft_topology_change_enabled(const db::config& cfg) {
+    return cfg.consistent_cluster_management() &&
+        cfg.check_experimental(db::experimental_features_t::feature::CONSISTENT_TOPOLOGY_CHANGES);
+}
+
+template <typename... T>
+static void initialize_system_keyspace_thread(sharded<db::system_keyspace>& sys_ks,
+        sharded<cql3::query_processor>& qp,
+        sharded<locator::snitch_ptr>& snitch,
```

Comment:
Note that `qp` and `snitch` are only needed to call `sys_ks.start()`. If you keep the call to `sharded<system_keyspace>::start()` in main() and make the `initialize_system_keyspace_thread()` be system_keyspace's method (as I suggested) the whole thing would get notably simpler and more self-contained.

Also note, that the latter change would make `db` argument unused too, as `system_keyspace` object already have `database&` reference on board.